components/raftstore/src/store/peer.rs

```diff
@@ -2332,6 +2356,8 @@ where
                         let _ = self.get_store().clear_data();
                         self.delay_clean_data = false;
                     }
+                    // Clear the uncampaigned list.
+                    self.uncampaigned_new_regions.0.clear();
```

Comment:
Is this line necessary? If the peer was not a follower before, it's uncampaigned_list must be empty — can we assert this? If so, it would also be helpful to add a line comment for`Peer::uncampaigned_new_regions` to clarify this key detail for better readability.

Reply:
When `split`, If the leader is missing, the follower who could not `campaign` should records the newly created regions into the uncampaigned list. And it should clear it after the new leader is elected.

Reply:
Please correct me if I'm wrong.
Assume we have a 3-nodes cluster where Peer1 is the current Leader. When splitting:
1. Peer1 transfer leader to the Peer2.
2. Peer2 enters the `StateRole::Leader` branch in `on_role_changed`, sends the Campaign messages, and clears the uncampaigned list.
3. Peer3 enters the `StateRole::Follower` branch in `on_role_changed`, also clears the uncampaigned list. Peer3 experiences a "role changed" event because `SoftState::leader_id` has changed, even though its StateRole remains the same.

Reply:
@LykxSassinator PTAL, thx

Reply:
> Is this line necessary? If the peer was not a follower before, it's uncampaigned list must be empty — can we assert this? If so, it would also be helpful to add a line comment for`Peer::uncampaigned_new_regions` to clarify this key detail for better readability.

This was my misunderstanding. Not only does the old leader go through this branch, but the other followers that weren't  promoted also do. So this line is for them.